stdlib/public/core/AssertCommon.swift

```diff
@@ -81,6 +81,10 @@ internal func _fatalErrorFlags() -> UInt32 {
 #endif
 }
 
+#if $Embedded
+public var assertLineNumber: UInt = 0
```

Comment:
I just tried out this patch, and under LTO, this *still* gets optimized out :) LTO is too powerful...

Reply:
Also, this should probably not be "public". Maybe something like this would make it not optimize away?

```
var _swift_assertLineNumber: UInt = 0
...
let p = UnsafePointer(&_swift_assertLineNumber)
Builtin.atomicstore_monotonic_volatile_Word(p._rawValue, line._builtinWordValue)
```